server/src/main/java/org/cloudfoundry/identity/uaa/health/HealthzEndpoint.java

```diff
@@ -45,8 +54,28 @@ public String getHealthz(HttpServletResponse response) {
             response.setStatus(503);
             return "stopping\n";
         } else {
-            return "ok\n";
+            if (wasLastConnectionSuccessful == null) {
+                return "No Check done.\n";
```

Comment:
Would this message cause confusion? Maybe it should mention that UAA is up but the database status is unknown.

Also, should this return 503 instead of 200 for this case?

Reply:
I would prefer to keep 200 here, as otherwise you could theoretically be very unlucky and the UAA may be restarted because of the error from /healthz, while the check was simply not yet done. 

I updated the message according to your feedback.